### PR TITLE
use eigen #19

### PR DESCRIPTION
**Pull request by ezrayst**

### CODE REVIEW

File: include/io.tcc [hunk 1]
```diff
@@ -18,8 +18,8 @@ void IO<Tdim>::write_material_points(
     //! Note that for 2D, z values are 0
     //! For 1D, both y and z values are 0
     for (auto const& points : materialpoints) {
-      for (double coordinate : points->coordinates()) {
-        material_points_file << coordinate << "\t";
+      for (unsigned i = 0; i < Tdim; ++i) {
```
Comment (kks32) [inline include/io.tcc]: Always iterate to the size of the container

File: include/io.tcc [hunk 2]
```diff
@@ -48,11 +47,11 @@ void IO<Tdim>::write_stresses(
     //! Stresses in Voigt Notation
     //! $\sigma_{xx}$ $\sigma_{yy}$ $\sigma_{zz}$
     //! $\tau_{yz}$ $\tau_{zx}$ $\tau_{xy}$
-    for (const auto& stress : stresses) {
+    for (auto const& stress : stresses) {
```
Comment (kks32) [inline include/io.tcc]: `const auto`

File: include/io.tcc [hunk 3]
```diff
@@ -48,11 +47,11 @@ void IO<Tdim>::write_stresses(
...
       stress_file.setf(std::ios::fixed, std::ios::floatfield);
       stress_file << id << '\t';
-      for (double stress_component : stress) {
-        stress_file << stress_component << '\t';
+      for (unsigned i = 0; i < Tdim * 2; ++i) {
```
Comment (kks32) [inline include/io.tcc]: Iterate to size of the container

**Review by kks32 — CHANGES_REQUESTED** [review]
Use `[]` instead of `()` in Eigen

**Comment by kks32** [comment]
@ezrayst Please fix circle CI integration and merge conflicts.

**Comment by ezrayst** [comment]
I am still not sure why it fails. I think it's because of my `<Eigen/Dense>` but why only some of them are having errors but not the others?

anyway it compiles and runs in my machine

File: CMakeLists.txt [hunk 4]
```diff
@@ -51,6 +51,7 @@ add_executable(mpm_point_generator ${mpm_point_gen_src})
 SET(test_src
   ${PROJECT_SOURCE_DIR}/tests/test.cc
   ${PROJECT_SOURCE_DIR}/tests/point_test.cc
+
```
Comment (kks32) [inline CMakeLists.txt]: Remove this space

File: include/gmsh.tcc [hunk 5]
```diff
@@ -4,139 +4,166 @@
...
+  Eigen::VectorXd elementarray(Tvertices);
+
+  //! specify element type 4 = tetrahedral, 5 = hexahedron
+  //! Documentation from GMSH
+  //! 1 - Line (2 nodes)
+  //! 2 - Triangle (3 nodes)
+  //! 3 - Quadrangle (4 nodes)
+  //! 4 - Tetrahedron (4 nodes)
+  //! 5 - Hexahedron (8 nodes)
+  //! 6 - Prism (6 nodes)
```
Comment (kks32) [inline include/gmsh.tcc]: Could you just add the relevant element type, and not all.

File: include/gmsh.tcc [hunk 6]
```diff
@@ -153,8 +180,8 @@ void GMSH<Tdim, Tvertices>::store_element_vertices() {
   const unsigned firstelement = elements_.begin()->first;
   const unsigned lastelement = elements_.rbegin()->first;
 
-  std::array<double, Tvertices> elementkeyvalues;
-  std::array<double, Tdim * Tvertices> verticesarray;
+  Eigen::VectorXd elementkeyvalues(Tvertices);
+  Eigen::VectorXd verticesarray(Tdim * Tvertices);
 
   //! Iterate through element_
   for (unsigned i = firstelement; i <= lastelement; ++i) {
```
Comment (kks32) [inline include/gmsh.tcc]: Why is it `<=` and can you not iterate over the `elements_` container as such using a range iterator?

File: include/gmsh.h [hunk 7]
```diff
@@ -1,16 +1,13 @@
...
 
-#include <array>
 #include <fstream>
 #include <iostream>
 #include <limits>
 #include <map>
 #include <memory>
 #include <vector>
 
-#include <Eigen/Dense>
```
Comment (kks32) [inline include/gmsh.h:12]: Don't you need `#include <eigen3/Eigen/Dense>`

**Review by kks32 — CHANGES_REQUESTED** [review]

File: include/gmsh.h (hunk 7 above)
Comment (ezrayst) [reply include/gmsh.h:12]: I have added them back. Well, all the header files include `point.h` so I thought we don't have to include this anymore

**Review by kks32 — APPROVED** [review]

**Pull request merged**